tests/vm_test.go

```diff
@@ -1921,6 +1924,63 @@ status:
 			}, 2*time.Minute, 1*time.Second).Should(Succeed())
 		})
 	})
+
+	Context("[Serial] when node becomes unhealthy", Serial, func() {
+		const componentName = "virt-handler"
+
+		getHandlerNodePod := func(nodeName string) *k8sv1.PodList {
+			pods, err := virtClient.CoreV1().Pods(flags.KubeVirtInstallNamespace).List(context.Background(),
+				metav1.ListOptions{
+					LabelSelector: fmt.Sprintf("kubevirt.io=%s", componentName),
+					FieldSelector: fmt.Sprintf("spec.nodeName=" + nodeName),
+				})
+			Expect(err).NotTo(HaveOccurred())
+			Expect(pods.Items).To(HaveLen(1))
+
+			return pods
+		}
+
+		It("[Serial] the VMs running in that node should be respawned", func() {
+
+			By("Starting VM")
+			vm := startVM(virtClient, createVM(virtClient, libvmi.NewCirros()))
+			vmi, err := virtClient.VirtualMachineInstance(vm.Namespace).Get(context.Background(), vm.Name, &k8smetav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+
+			nodeName := vmi.Status.NodeName
+			oldUID := vmi.UID
+
+			By("Blocking virt-handler from reconciling the VMI")
+			libpod.AddKubernetesApiBlackhole(getHandlerNodePod(nodeName), componentName)
```

Comment:
nit: You could potentially save the result of `getHandlerNodePod(nodeName)` in a var.

Reply:
Done.